frb_codegen/src/generator/dart/mod.rs

```diff
@@ -221,18 +251,23 @@ fn generate_dart_declaration_body(
     )
 }
 
-fn generate_dart_implementation_body(
-    dart_funcs: &[GeneratedApiFunc],
-    dart_api2wire_funcs: &[String],
-    dart_api_fill_to_wire_funcs: &[String],
-    dart_wire2api_funcs: &[String],
-    dart_api_impl_class_name: &str,
-    dart_wire_class_name: &str,
-    dart_api_class_name: &str,
-) -> String {
+fn generate_dart_implementation_body(spec: &DartApiSpec, config: &Opts) -> String {
+    let dart_api_impl_class_name = config.dart_api_impl_class_name();
+    let dart_wire_class_name = config.dart_wire_class_name();
+    let dart_api_class_name = config.dart_api_class_name();
+    let DartApiSpec {
+        dart_funcs,
+        dart_api2wire_funcs,
+        dart_api_fill_to_wire_funcs,
+        dart_wire2api_funcs,
+        dart_wasm_funcs,
+        dart_wasm_module,
+        ..
+    } = spec;
+    let wasm = config.wasm;
     format!(
         "class {dart_api_impl_class_name} extends FlutterRustBridgeBase<{dart_wire_class_name}> implements {dart_api_class_name} {{
-            factory {dart_api_impl_class_name}(ffi.DynamicLibrary dylib) => {dart_api_impl_class_name}.raw({dart_wire_class_name}(dylib));
+            factory {dart_api_impl_class_name}({external_lib_type} dylib) => {dart_api_impl_class_name}.raw({dart_wire_class_name}(dylib));
```

Comment:
Good job - we should have used named arguments in `format!` instead of positional arguments before... Maybe we can refactor other legacy code later, or make an issue labeled `good-first-issue`.